ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala

```diff
@@ -26,6 +27,128 @@ import io.grpc.{Metadata, StatusRuntimeException}
 import scalaz.syntax.tag._
 
 class ErrorFactories private (errorCodesVersionSwitcher: ErrorCodesVersionSwitcher) {
+
+  object TrackerErrors {
+
+    object QueueSubmitFailure {
+
+      def failedToEnqueue(t: Throwable)(implicit
+          contextualizedErrorLogger: ContextualizedErrorLogger
+      ): Status = {
+        errorCodesVersionSwitcher.choose(
+          v1 = {
+            val status = io.grpc.Status.ABORTED
+              .withDescription(s"Failed to enqueue: ${t.getClass.getSimpleName}: ${t.getMessage}")
+              .withCause(t)
+            val statusBuilder = GrpcStatus.toJavaBuilder(status)
+            GrpcStatus.buildStatus(Map.empty, statusBuilder)
+          },
+          v2 = LedgerApiErrors.InternalError
+            .CommandTrackerInternalError(
+              message = s"Failed to enqueue: ${t.getClass.getSimpleName}: ${t.getMessage}",
+              throwableO = Some(t),
+            )
+            .asGrpcStatusFromContext,
+        )
+      }
+
+      def failed(t: Throwable)(implicit
+          contextualizedErrorLogger: ContextualizedErrorLogger
+      ): Status = {
+        errorCodesVersionSwitcher.choose(
+          v1 = {
+            val status = io.grpc.Status.ABORTED
+              .withDescription(s"Failure: ${t.getClass.getSimpleName}: ${t.getMessage}")
+              .withCause(t)
+            val statusBuilder = GrpcStatus.toJavaBuilder(status)
+            GrpcStatus.buildStatus(Map.empty, statusBuilder)
+          },
+          v2 = LedgerApiErrors.CommandErrors.CommandSubmissionFailure
+            .Reject(
+              messagePrefix = "Failure",
+              throwableO = Some(t),
+            )
+            .asGrpcStatusFromContext,
+        )
+      }
+
+      def ingressBufferFull()(implicit
+          contextualizedErrorLogger: ContextualizedErrorLogger
+      ): Status = {
+        errorCodesVersionSwitcher.choose(
+          v1 = {
+            val status = io.grpc.Status.RESOURCE_EXHAUSTED
+              .withDescription("Ingress buffer is full")
+            val statusBuilder = GrpcStatus.toJavaBuilder(status)
+            GrpcStatus.buildStatus(Map.empty, statusBuilder)
+          },
+          v2 = SubmissionErrors.ParticipantBackpressure
+            .Rejection("Ingress buffer is full")
```

Comment:
Let's add a bit more context
```suggestion
            .Rejection("Command service ingress buffer is full")
```

Reply:
Ok